### PR TITLE
Feature/18249 ncatools identifier

### PR DESCRIPTION
**Pull request by danwilliams**

### CODE REVIEW

File: admin/spec/service/nca-tools-user.service.spec.js [hunk 1]
```diff
@@ -115,51 +115,4 @@ describe('nca-tools-user.service', () => {
       expect(userDataService.sqlUpdateSchool).not.toHaveBeenCalled()
     })
   })
-
-  describe('recordLogonAttempt', () => {
```
Comment (GuyHarwood) [inline admin/spec/service/nca-tools-user.service.spec.js]: just so i understand the bigger picture here - is this recorded elsewhere, in adminLogonEvent for example?

File: admin/services/nca-tools-user.service.js [hunk 2]
```diff
@@ -49,12 +49,6 @@ const service = {
     userRecord.mtcRole = roleService.mapNcaRoleToMtcRole(ncaUser.UserType, school)
     return userRecord
   },
-  recordLogonAttempt: async (logonData) => {
```
Comment (GeorgeChatzigiannis) [inline admin/services/nca-tools-user.service.js]: is this supposed to be removed?

File: admin/services/nca-tools-user.service.js [hunk 3]
```diff
@@ -29,7 +28,8 @@ const service = {
       const mtcRoleName = roleService.mapNcaRoleToMtcRole(ncaUser.UserType, school)
       const role = await roleService.findByTitle(mtcRoleName)
       const user = {
-        identifier: ncaUser.EmailAddress,
+        identifier: ncaUser.externalAuthenticationId,
+        displayName: ncaUser.EmailAddress,
```
Comment (jon-shipley) [inline admin/services/nca-tools-user.service.js]: This doesn't seem quite right.   Shouldn't the display name be a name, e.g. `ncaUser.UserName`?

Comment (GuyHarwood) [reply admin/services/nca-tools-user.service.js]: We don't actually care what the email address is, and it sometimes isnt an actual email address! (go figure).

We do however store the contents of that field in the top right of the admin header, so we are accomodating this with the 'displayName' field.

Comment (GuyHarwood) [reply admin/services/nca-tools-user.service.js]: btw - in the payload i think its actually called `userName`

File: admin/spec/service/nca-tools-user.service.spec.js (hunk 1 above)
Comment (danwilliams) [reply admin/spec/service/nca-tools-user.service.spec.js]: The recordLogonAttempt was the only piece of code using ncaToolsSession, and there is no code that calls it. I believe it is recorded in admin-logon-event (as in `authentication/local-strategy.js` @98-115 and `authentication/nca-tools-authentication-strategy.js` @52 and @60).

File: admin/services/nca-tools-user.service.js (hunk 2 above)
Comment (danwilliams) [reply admin/services/nca-tools-user.service.js]: Please see my response to Guy's question about `admin/spec/service/nca-tools-user.service.spec.js` above.

File: admin/spec/service/nca-tools-user.service.spec.js (hunk 1 above)
Comment (GuyHarwood) [reply admin/spec/service/nca-tools-user.service.spec.js]: this is what i assumed to be the case.

File: admin/services/nca-tools-user.service.js (hunk 3 above)
Comment (danwilliams) [reply admin/services/nca-tools-user.service.js]: So, just to be sure, is the displayName okay, or do we want the UserName instead? (The spec was saying "populate users.displayName field with email address").

Comment (GuyHarwood) [reply admin/services/nca-tools-user.service.js]: IIRC email address is always populated, username is not.  leave as is for now

Comment (jon-shipley) [reply admin/services/nca-tools-user.service.js]: What kind of auth service doesn't give us the real name?

**Pull request merged**